File: cesar.c
#include <stdio.h>

void cesar(char *string, int i){
if(string[i] == '\0') return; // se chegar no final da string acaba a recursividade

if (string[i] != '\n') {
        string[i] = string[i] + 3; // atualiza o char
    }
cesar(string,i+1); // chama recursivamente pro proxima char
}
int main(){
    char string[10000];
    while(fgets(string, 10000,stdin) != EOF){
        if(string[0] == 'F' && string[1] && 'I' && string[2] && 'M' && string[3] == '\n') return 0; // verifica se é FIM se for retorna 
    
    
    cesar(string,0); // passa I = 0 pra começar da string[0]
    printf("%s", string);
}
return 0;
}

File: cesar.java
import java.util.Scanner;

public class cesar {
    static boolean verificar(String s){
        if (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M') return false;
        return true;
    } // verifica se a palavra é FIM
    static String novastring(String s) {
    StringBuilder sb = new StringBuilder();
    
    for (int i = 0; i < s.length(); i++) {
        // Pega o caractere, soma 3 e transforma de volta em char
        char novoChar = (char) (s.charAt(i) + 3);
        sb.append(novoChar);
    }
    
    return sb.toString(); // Retorna a nova string montada
}
public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    while(sc.hasNextLine()){     // roda enquanto tem linha pra ler
    String s = sc.nextLine();
    if(verificar(s) != true) return; // chama pra verificar se é FIM
    String s1 = novastring(s); // Cria a nova string de cesar
    System.out.println(s1); 
    }
}
}
